File: others/xorshift/xorshift.jl
RNGState::Int32 = 0;

function xorshift()
    global RNGState ⊻= RNGState << 13
    global RNGState ⊻= RNGState >> 17
    global RNGState ⊻= RNGState << 5
    return RNGState
end

if length(ARGS) < 2
    println("Not enough arguments provided")
    println("Usage:")
    println("  <seed> <iterations>")
    println("Description:")
    println("  Seeds the RNG with `seed` and run xorshift `iterations` many times.")
    exit(1)
end

RNGState = parse(Int32, ARGS[1]);
iters = parse(Int32, ARGS[2]);

for _ in 1:iters
    xorshift()
end

println(RNGState);

File: others/xorshift/xorshift.c
#include <stdio.h> // For printf
#include <stdlib.h> // For exit

int RNGState = 0;

int xorshift()
{
	RNGState ^= RNGState << 13;
	RNGState ^= RNGState >> 17;
	RNGState ^= RNGState << 5;
	return RNGState;
}

int main(int argc, char const *argv[])
{
	if (argc < 3)
	{
		printf("Not enough arguments provided\n");
		printf("Usage:\n");
		printf("  <seed> <iterations>\n");
		printf("Description:\n");
		printf("  Seeds the RNG with `seed` and run xorshift `iterations` many times.\n");
		exit(1);
	}

	RNGState = atoi(argv[1]);
	int iters = atoi(argv[2]);

	for (int i = 0; i < iters; i++) xorshift();

	printf("%d\n", RNGState);

	return 0;
}